ros2bag/ros2bag/verb/record.py

```diff
@@ -122,41 +79,36 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             help='record also hidden topics.'
         )
         parser.add_argument(
-            '--qos-profile-overrides-path', type=argparse.FileType('r'),
+            '--qos-profile-overrides-path', type=FileType('r'),
             help='Path to a yaml file defining overrides of the QoS profile for specific topics.'
         )
         self._subparser = parser
 
-    def create_bag_directory(self, uri):
-        try:
-            os.makedirs(uri)
-        except OSError:
-            return "[ERROR] [ros2bag]: Could not create bag folder '{}'.".format(uri)
-
     def main(self, *, args):  # noqa: D102
         if args.all and args.topics:
-            return 'Invalid choice: Can not specify topics and -a at the same time.'
+            return print_error('Invalid choice: Can not specify topics and -a at the same time.')
 
         uri = args.output or datetime.datetime.now().strftime('rosbag2_%Y_%m_%d-%H_%M_%S')
 
         if os.path.isdir(uri):
-            return "[ERROR] [ros2bag]: Output folder '{}' already exists.".format(uri)
+            return print_error("Output folder '{}' already exists.".format(uri))
```

Comment:
Consider moving the argument validation in the `arpgarse` logic if you keep modifying this code. This will take care for you of reporting errors to the users the "proper way". You could just create a function checking if the directory exists, and call `strftime` to pass a default value as it's built here.